### DISCUSSION TITLE
GitHub App to send client_payload to trigger a workflow with repository_dispatch event

### DISCUSSION THREAD

**Discussion by Keshapally** [community/community#62659] [Apps, API and Webhooks] [answered]
### Select Topic Area

General

### Body

Hi All,

I'm trying to trigger a GitHub Workflow when events like someone add a topic, issue, label in their repositories in our ORG.
I've created a GitHub App with required settings (Issue comments, Labels, Issues, Repositories).

Now, how can i send payload to workflow in this specific Body parameters which is mentioned  here ?
 https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-a-repository-dispatch-event

I'm suspecting that I'm unable to trigger my workflow since webhook/githubapp payload format is not accepted by Repository_dispatch event.

Please suggest.

Thank you :)

**Comment by Keshapally** [comment]
Looks like this https://github.com/orgs/community/discussions/50955 was unanswered :(

**Comment by Yash-Dobariya** [comment]
To trigger a GitHub Actions workflow using the repository_dispatch event, you need to send a POST request to the GitHub API with a specific payload. The payload is a JSON object that can include custom data and parameters for your workflow.

Based on the documentation you provided, the payload should be sent as the request body with the appropriate headers set. The API endpoint you need to use is:
POST /repos/{owner}/{repo}/dispatches

curl -X POST \
  -H "Authorization: token YOUR_GITHUB_APP_TOKEN" \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/{owner}/{repo}/dispatches \
  -d '{"event_type": "your-custom-event-name", "client_payload": {"key1": "value1", "key2": "value2"}}'

Replace YOUR_GITHUB_APP_TOKEN with the actual token of your GitHub App. Make sure to replace {owner} and {repo} with the appropriate organization and repository names, respectively.

In the payload, "event_type" is a required field that represents the custom event name you want to trigger in your workflow. You can also include "client_payload" with additional data if needed.

Once you send this POST request, GitHub will trigger the workflow associated with the repository_dispatch event in the repository, and you can use the payload data in your workflow as needed.

Make sure your GitHub App has appropriate permissions and access to the repositories to trigger the workflow successfully. Also, ensure that your workflow is correctly set up to listen to the repository_dispatch event.

If you're still facing issues with triggering your workflow, make sure to check the GitHub App's permissions and the repository's settings. Additionally, review the workflow configuration to ensure that it is properly configured to respond to the repository_dispatch event.

**Reply by Keshapally** [reply]
Thanks @Yash-Dobariya for your response on this.

Yes, I was able to trigger my workflow from postman using POST API & repository_dispatch.
Because in API tool i was able to customize JSON Body with  "event_type"  & "client_payload"
Whereas in GitHub webhook/App i don't see an option to do that , it only sends payload.


Why ? : I need GitHub webhook/App to listen to Repo events on repositories & trigger workflow along with payload

**Reply by Yash-Dobariya** [reply, marked as answer]
GitHub Apps and webhooks have different functionalities. Webhooks are specifically designed to listen to GitHub events and send data to external endpoints (e.g., your server or service) when those events occur. They are not designed to send specific payloads to trigger GitHub Actions workflows directly.

GitHub Apps, on the other hand, are more powerful and can interact directly with GitHub repositories, perform actions, and respond to various events. When you want to trigger a GitHub Actions workflow with a custom payload, as in the case of the repository_dispatch event, you need to use an API call to achieve this.

**Comment by Tomx999** [comment]
Z

**Reply by ramyanarsupalli** [reply]
You mean 
 YOUR_GITHUB_APP_TOKEN  --> the pem key?
